src/Bicep.LangServer/Snippets/ResourceSnippetsProvider.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+namespace Bicep.LanguageServer.Snippets
+{
+    public class ResourceSnippetsProvider : IResourceSnippetsProvider
+    {
+        private static readonly Dictionary<string, string> _snippetNameToDetailMap = new Dictionary<string, string>()
+        {
+            { "res-aks-cluster", "Kubernetes Service Cluster" },
+            { "res-app-security-group", "Application Security Group" },
+            { "res-automation-account", "Automation Account" },
+            { "res-availability-set", "Availability Set" },
+            { "res-container-group", "Container Group" },
+            { "res-container-registry", "Container Registry" },
+            { "res-cosmos-account", "Cosmos DB Database Account" },
+            { "res-data-lake", "Data Lake Store Account" },
+            { "res-dns-zone", "DNS Zone" },
+            { "res-ip", "Public IP Address"},
+            { "res-ip-prefix", "Public IP Prefix"}
+        };
+
+        private HashSet<ResourceSnippet> resourceSnippets = new HashSet<ResourceSnippet>();
+
+        public ResourceSnippetsProvider()
+        {
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            string? currentDirectory = Path.GetDirectoryName(Assembly.GetEntryAssembly()?.Location);
+            string templatesFolder = Path.Combine(currentDirectory!, "Snippets", "Templates");
```

Comment:
>! [](start = 66, length = 1)

Our rule is to bend over backwards to avoid null suppression on the product side of things. In this case you could add the following to the above line `?? throw new NotImplementedException(...)` or something similar. In test code `!` is fair game.

Reply:
Updated as part of below commit. Thanks!

https://github.com/Azure/bicep/commit/91ec6ff34132d0b6be98410a28548453a5b15375